src/app/services/BadgeService/badge.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../../environments/environment';
+import { APIService } from '../api.service';
+import 'rxjs/Rx';
+import { catchError, map, tap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { BadgeConfig } from '../../badge/Config';
+import { Badge } from '../../badge/models/badgeModel';
+import { NotifyService } from '../../badge/notify';
+
+@Injectable()
+
+export class BadgeService {
+
+  private _baseAPIUrl = environment.baseURL + environment.apiEndpoint;
+  private _getUserBadge = this._baseAPIUrl + 'user/badge';
+  private _updateUserBadge = this._baseAPIUrl + 'user/badge/update';
+
+
+  constructor(private _apiService: APIService, private _notify: NotifyService) {
+  }
+
+  getBadge(): Observable<Badge> {
```

Comment:
Do we have any unit test cases foe these services?

Reply:
Added.